x-pack/plugins/ml/public/application/memory_usage/memory_tree_map/tree_map.tsx

```diff
@@ -0,0 +1,188 @@
+import { useTrainedModelsApiService } from '../../services/ml_api_service/trained_models';
+import { LoadingWrapper } from '../../jobs/new_job/pages/components/charts/loading_wrapper';
+import { useFieldFormatter } from '../../contexts/kibana';
+
+import { useRefresh } from '../../routing/use_refresh';
+import { getMemoryItemColor } from '../memory_item_colors';
+
+interface Props {
+  node?: string;
+  type?: MlSavedObjectType;
+  height?: string;
+}
+
+const DEFAULT_CHART_HEIGHT = '400px';
+
+const TYPE_OPTIONS: TreeMapOptions[] = [
+  {
+    label: i18n.translate('xpack.ml.memoryUsage.treeMap.adLabel', {
+      defaultMessage: 'Anomaly detection jobs',
+    }),
+    objectType: 'anomaly-detector',
+    color: getMemoryItemColor('anomaly-detector'),
+  },
+  {
+    label: i18n.translate('xpack.ml.memoryUsage.treeMap.dfaLabel', {
+      defaultMessage: 'Data frame analytics jobs',
+    }),
+    objectType: 'data-frame-analytics',
+    color: getMemoryItemColor('data-frame-analytics'),
+  },
+  {
+    label: i18n.translate('xpack.ml.memoryUsage.treeMap.modelsLabel', {
+      defaultMessage: 'Trained models',
+    }),
+    objectType: 'trained-model',
+    color: getMemoryItemColor('trained-model'),
+  },
+];
+
+type TreeMapOptions = EuiComboBoxOptionOption & { objectType: MlSavedObjectType };
+
+export const JobMemoryTreeMap: FC<Props> = ({ node, type, height }) => {
+  const bytesFormatter = useFieldFormatter(FIELD_FORMAT_IDS.BYTES);
+  const refresh = useRefresh();
+  const chartHeight = height ?? DEFAULT_CHART_HEIGHT;
+
+  const trainedModelsApiService = useTrainedModelsApiService();
+  const [allData, setAllData] = useState<MemoryUsageInfo[]>([]);
+  const [data, setData] = useState<MemoryUsageInfo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [selectedOptions, setSelectedOptions] = useState<TreeMapOptions[]>(TYPE_OPTIONS);
+
+  const filterData = useCallback(
+    (dataIn: MemoryUsageInfo[]) => {
+      const labels = selectedOptions.map((o) => o.objectType);
+      return dataIn.filter((d) => labels.includes(d.type));
+    },
+    [selectedOptions]
+  );
+
+  const onTypeChange = useCallback((types: EuiComboBoxOptionOption[]) => {
+    setSelectedOptions(types as TreeMapOptions[]);
+  }, []);
+
+  const loadJobMemorySize = useCallback(async () => {
+    setLoading(true);
+    const resp = await trainedModelsApiService.memoryUsage(type, node);
+    setAllData(resp);
+    setLoading(false);
+  }, [trainedModelsApiService, type, node]);
+
+  useEffect(
+    function redrawOnFilterChange() {
+      setData(filterData(allData));
+    },
+    [selectedOptions, allData, filterData]
+  );
+
+  useEffect(
+    function updateOnTimerRefresh() {
+      loadJobMemorySize();
+    },
+    [loadJobMemorySize, refresh]
+  );
+
+  return (
+    <div
+      style={{ height: chartHeight }}
+      data-test-subj={`mlJobTreeMap ${data.length ? 'withData' : 'empty'}`}
```

Comment:
I guess would be better like this to preserve a camelCase notation 
```suggestion
      data-test-subj={`mlJobTreeMap ${data.length ? 'WithData' : 'Empty'}`}
```

Reply:
There is a space after `mlJobTreeMap`.
I copied the format which is used [here](https://github.com/elastic/kibana/blob/64e7c05800e8a1903d4ecb38b9d511343a8185b6/x-pack/plugins/ml/public/application/jobs/new_job/pages/components/charts/event_rate_chart/event_rate_chart.tsx#L64) which is used in tests [here](https://github.com/elastic/kibana/blob/64e7c05800e8a1903d4ecb38b9d511343a8185b6/x-pack/test/functional/services/ml/job_wizard_common.ts#L95)

For the sake of consistency I'll leave this as is, and possibly change it and the original when I write the functional tests.